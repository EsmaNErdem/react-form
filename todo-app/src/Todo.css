.Todo-Completed {
    text-decoration: line-through;
}