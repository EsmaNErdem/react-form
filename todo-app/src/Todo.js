import React, { useState } from "react";
import "./Todo.css"

const Todo = ({id, task, handleRemove, handleUpdate}) => {
    const [isEditting, setIsEditting] = useState(false)
    const [edit, setEdit] = useState(task)
    const [markCompleted, setMarkCompleted] = useState(false)

    const toggleEdit = () => {
        setIsEditting(isEditting => !isEditting)
    }

    const handleSubmit = e =>{
        e.preventDefault()
        handleUpdate(id, edit)
        setIsEditting(false)
    }

    const handleChange = e => {
        setEdit(e.target.value)
    }

    const toggleCompleted = () => {
        setMarkCompleted(mark => !mark)
    }

    let toDo = (
        <div>
        <li className={markCompleted ? "Todo-Completed" : ""}>{task}</li>
        <button  onClick={() => handleRemove(id)}>X</button>
        <button onClick={toggleEdit}>Edit</button>
        <button onClick={toggleCompleted}>
        {markCompleted ? "Reverse Mark Completed" : "Mark Completed"}
        </button>
      </div>
    )

    if (isEditting) {
        toDo = (
        <form onSubmit={handleSubmit}>
            <label htmlFor="taskEdit">Edit Todo Task:</label>
            <input
                id="taskEdit"
                name="task"
                type="text"
                value={edit}
                onChange={handleChange}
            />
            <button>Update task!</button>
        </form>
        )
    }

    return toDo
}

export default Todo;