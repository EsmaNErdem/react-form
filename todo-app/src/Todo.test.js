import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

test("renders without crashing", () => {
    render(<Todo />);
});

test("it renders and matches with snaphot", () => {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});


test("it renders and matches with snaphot when editting", () => {
    const { asFragment, getByText } = render(<Todo />);
    const editBtn = getByText("Edit")
    fireEvent.click(editBtn)
    expect(asFragment()).toMatchSnapshot();
});