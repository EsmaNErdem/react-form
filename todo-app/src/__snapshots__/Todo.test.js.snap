// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches with snaphot 1`] = `
<DocumentFragment>
  <div>
    <li
      class=""
    />
    <button>
      X
    </button>
    <button>
      Edit
    </button>
    <button>
      Mark Completed
    </button>
  </div>
</DocumentFragment>
`;

exports[`it renders and matches with snaphot when editting 1`] = `
<DocumentFragment>
  <form>
    <label
      for="taskEdit"
    >
      Edit Todo Task:
    </label>
    <input
      id="taskEdit"
      name="task"
      type="text"
      value=""
    />
    <button>
      Update task!
    </button>
  </form>
</DocumentFragment>
`;
